executor/runner/runner.go

```diff
@@ -368,7 +368,7 @@ func (r *Runner) maybeSetupExternalLogger(ctx context.Context, logDir string) er
 	}
 	logger.G(ctx).Info("Starting external logger")
 
-	wConf := filesystems.NewWatchConfig(logDir, r.container.UploadDir("logs"), r.container.LogUploadRegexp(), *r.container.LogUploadCheckInterval(), *r.container.LogUploadThresholdTime(), *r.container.LogStdioCheckInterval(), r.container.LogKeepLocalFileAfterUpload())
+	wConf := filesystems.NewWatchConfig(logDir, r.container.UploadDir("logs"), r.container.LogUploadRegexp(), *r.container.LogUploadCheckInterval(), *r.container.LogUploadThresholdTime(), *r.container.LogStdioCheckInterval(), r.container.LogKeepLocalFileAfterUpload(), r.container.FinalUploadMatchRegexp())
```

Comment:
What is the difference between `r.container.LogUploadRegexp()` and `r.container.FinalUploadMatchRegexp()`?

Reply:
It's a regex, which answers the question 'should I only upload files that match regex in final upload'? Defaults to false.

Reply:
I was also confused by this variable. What if was renamed to `UploadLogsBasedOnRotateRegexp`
(Although maybe it hints that there should just be 2 regexes, one for rotate (`defaultUploadRegexpList`)and one for upload (/logs/.*), and in this case users would want to update their upload one?)

Reply:
Not to pile on too much, but I also got confused by this. :)

I _think_ it might read more clearly if this was another Regexp - `FinalUploadRegexp()`.  If the agent param is set, then this is set to `LogUploadRegexp()`.  Otherwise, it defaults to `.*`.  I'm a bit on the fence about how necessary this is, though.